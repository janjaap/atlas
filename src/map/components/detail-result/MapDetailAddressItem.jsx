import React from 'react';
import PropTypes from 'prop-types';

const MapDetailAddressItem = ({ label, values }) => (
  <div className="map-detail-result__item">
    <section className="map-detail-result__item-content">
      <div className="map-detail-result__item-label">{label}</div>
      <div className="map-detail-result__item-value">
        <div>{values.straatnaam} {values.huisnummer} {values.huisnummertoevoeging}</div>
        <div>{values.postcode} {values.plaats}</div>
      </div>
    </section>
  </div>
);

MapDetailAddressItem.propTypes = {
  label: PropTypes.string.isRequired,
  values: PropTypes.shape({
    huisnummer: PropTypes.string,
    huisnummertoevoeging: PropTypes.string,
    plaats: PropTypes.string,
    postcode: PropTypes.string,
    straatnaam: PropTypes.string
  }).isRequired
};

export default MapDetailAddressItem;
