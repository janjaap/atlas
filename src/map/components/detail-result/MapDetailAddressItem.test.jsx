import React from 'react';
import { shallow } from 'enzyme';

import MapDetailAddressItem from './MapDetailAddressItem';

describe('MapDetailAddressItem', () => {
  it('should render without huisnummertoevoeging', () => {
    const label = 'Address label';
    const values = {
      huisnummer: '1515',
      plaats: 'Amsterdam',
      postcode: '1012MT',
      straatnaam: 'Nieuwendijk'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAddressItem
        label={label}
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        values={values}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with huisnummertoevoeging', () => {
    const label = 'Address label';
    const values = {
      huisnummer: '1515',
      huisnummertoevoeging: 'bis',
      plaats: 'Amsterdam',
      postcode: '1012MT',
      straatnaam: 'Nieuwendijk'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAddressItem
        label={label}
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        values={values}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
