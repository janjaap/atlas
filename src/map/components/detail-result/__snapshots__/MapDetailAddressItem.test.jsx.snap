// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailAddressItem should render with huisnummertoevoeging 1`] = `
<div
  className="map-detail-result__item"
>
  <section
    className="map-detail-result__item-content"
  >
    <div
      className="map-detail-result__item-label"
    >
      Address label
    </div>
    <div
      className="map-detail-result__item-value"
    >
      <div>
        Nieuwendijk
         
        1515
         
        bis
      </div>
      <div>
        1012MT
         
        Amsterdam
      </div>
    </div>
  </section>
</div>
`;

exports[`MapDetailAddressItem should render without huisnummertoevoeging 1`] = `
<div
  className="map-detail-result__item"
>
  <section
    className="map-detail-result__item-content"
  >
    <div
      className="map-detail-result__item-label"
    >
      Address label
    </div>
    <div
      className="map-detail-result__item-value"
    >
      <div>
        Nieuwendijk
         
        1515
         
      </div>
      <div>
        1012MT
         
        Amsterdam
      </div>
    </div>
  </section>
</div>
`;
